src/binaryen-c.cpp

```diff
@@ -299,12 +299,13 @@ BinaryenOp BinaryenCurrentMemory(void) { return CurrentMemory; }
 BinaryenOp BinaryenGrowMemory(void) { return GrowMemory; }
 BinaryenOp BinaryenHasFeature(void) { return HasFeature; }
 
-BinaryenExpressionRef BinaryenBlock(BinaryenModuleRef module, const char* name, BinaryenExpressionRef* children, BinaryenIndex numChildren) {
+BinaryenExpressionRef BinaryenBlock(BinaryenModuleRef module, const char* name, BinaryenExpressionRef* children, BinaryenIndex numChildren, BinaryenType type) {
   auto* ret = ((Module*)module)->allocator.alloc<Block>();
   if (name) ret->name = name;
   for (BinaryenIndex i = 0; i < numChildren; i++) {
     ret->list.push_back((Expression*)children[i]);
   }
+  if (type) ret->type = WasmType(type);
   ret->finalize();
```

Comment:
actually there is something ambiguous here. if type is 0, that means the none type, and we could also call `finalize(none)`. but that means that there is no way for the user to say "figure out the type" (the old way) as opposed to "i am giving you the type". As the code is written right now, 0 means "figure out the type" instead of "none".

I guess we could add an api for making blocks without providing the type. but maybe it's fine to just change this to do `finalize(type)` regardless of the value of `type`. Thoughts?